### DISCUSSION TITLE
Understanding the codebase

### DISCUSSION THREAD

**Discussion by nalin-singh** [Bhupesh-V/ugit#37] [Q&A] [answered]
So I wanted to contribute to the undo git rm functionality but how does a person really understand the current codebase because it's not straight forward javascript or something. Anything to look for to understand the language ?

**Comment by Bhupesh-V** [comment, marked as answer]
Hey @nalin-singh Thanks for asking this question.
The language utilized for writing ugit is `bash` (or more specifically, it's a shell script). You can have a quick overview of bash [here](https://learnxinyminutes.com/docs/bash/). You will be able to find plenty of guides on **How to learn shell scripting** online as well.

Hope this answers your question
